extensions/standard-processors/processors/GetFile.cpp

```diff
@@ -146,30 +146,34 @@ void GetFile::onSchedule(core::ProcessContext *context, core::ProcessSessionFact
   if (context->getProperty(FileFilter.getName(), value)) {
     request_.fileFilter = value;
   }
+
+  if (!context->getProperty(Directory.getName(), value)) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Input Directory property is missing");
+  }
+  if (!utils::file::FileUtils::is_directory(value.c_str())) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Input Directory \"" + value + "\" is not a directory");
+  }
 }
 
 void GetFile::onTrigger(core::ProcessContext *context, core::ProcessSession *session) {
   // Perform directory list
 
   metrics_->iterations_++;
 
-  logger_->log_debug("Is listing empty %i", isListingEmpty());
-  if (isListingEmpty()) {
+  const bool isDirEmptyBeforePoll = isListingEmpty();
+  logger_->log_debug("Is listing empty before polling directory %i", isDirEmptyBeforePoll);
+  if (isDirEmptyBeforePoll) {
     if (request_.pollInterval == 0 || (getTimeMillis() - last_listing_time_) > request_.pollInterval) {
       std::string directory;
-      const std::shared_ptr<core::FlowFile> flow_file;
-      if (context->getProperty(Directory, directory, flow_file)) {
-        logger_->log_warn("Resolved missing Input Directory property value");
-        performListing(directory, request_);
-        last_listing_time_.store(getTimeMillis());
-      } else {
-        return;
-      }
+      performListing(directory, request_);
```

Comment:
This feels bad to me, directory is an empty string here. 

Even though the property is queried in onSchedule, I don't see the value saved to a member anywhere. (Maybe I missed something)

Reply:
Oops, my bad, unit test also got this.